lib/ruby_lsp/listeners/completion.rb

```diff
@@ -128,28 +128,30 @@ def on_constant_path_node_enter(node)
 
       sig { params(node: Prism::CallNode).void }
       def on_call_node_enter(node)
-        receiver = node.receiver
-
-        # When writing `Foo::`, the AST assigns a method call node (because you can use that syntax to invoke singleton
-        # methods). However, in addition to providing method completion, we also need to show possible constant
-        # completions
-        if (receiver.is_a?(Prism::ConstantReadNode) || receiver.is_a?(Prism::ConstantPathNode)) &&
-            node.call_operator == "::"
-
-          name = constant_name(receiver)
-
-          if name
-            start_loc = node.location
-            end_loc = T.must(node.call_operator_loc)
-
-            constant_path_completion(
-              "#{name}::",
-              Interface::Range.new(
-                start: Interface::Position.new(line: start_loc.start_line - 1, character: start_loc.start_column),
-                end: Interface::Position.new(line: end_loc.end_line - 1, character: end_loc.end_column),
-              ),
-            )
-            return
+        unless @global_state.has_type_checker
```

Comment:
I think we can check if `sorbet_level` is `Disabled` for this case?

Basically, `@global_state.has_type_checker` won't be needed anymore.

Reply:
We still need `@global_state.has_type_checker` to prevent registering for workspace symbols (since that's workspace wide and not document specific), but indeed we don't need `has_type_checker` for any text document requests anymore.